### PR TITLE
Adding support to Stop execution if an exception is thrown when adding a custom domain to a website. #10422 

### LINKED ISSUE #10422: webapp: Stop executing if an exception is thrown when adding a custom domain to a website.
## Description
When a user makes a mistake in their powershell script. It sometimes results in all their existing custom domains being removed. As part of the WebsiteClient in chaning customDomains the client does a GET to see existing domains, adds new domains, and then Deletes domains that are not included. If a customDomain fails to be added, we should not remove custom domains and should stop that part of the update customDomains code.

## Steps to reproduce
I understand that this isnt the newest version, but I forgot to save those commands
```powershell
$W=Get-AzureRmWebApp -ResourceGroupName "prod-rg" -Name "prod-rg"
$X = $W.hostnames
$X.add("mycustom.domain.com")
$HN = $X.tostring()
Set-AzureRmWebApp -ResourceGroupName "prod-rg" -Name "prod-rg" -HostNames  $HN

```
At this point, X = [oldDomain1, OldDomain2, New domain]
And HN = system.collections.generic.list`1[system.string]
When customer tries to send HN the websites cmdlet sends that one new domain, this fails because this 'domain' is not valid. and then the websites cmdlet continues 
## Environment data

```

```

## Module versions

```powershell

```

## Debug output

```

```

## Error output

```

```

### PR DESCRIPTION
**Pull request by ThejaChoudary**
Fixes #10422

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by panchagnula** [comment]
@ThejaChoudary the linked GitHub issue has repro steps can we use this & add as a test case to make sure this scenario is covered to make sure we have no future regressions?

File: src/Websites/Websites/Utilities/WebsitesClient.cs [hunk 1]
```diff
@@ -145,6 +146,7 @@ public void AddCustomHostNames(string resourceGroupName, string location, string
                             {
                                 SiteName = webAppName,
                             });
+                        successCount++;
```
Comment (panchagnula) [inline src/Websites/Websites/Utilities/WebsitesClient.cs]: I don't think we need to keep track of successcount , the problem here is that the hostname value is an invalid format from the example in the issue, the value is "system.collections.generic.list`1[system.string]" as a result the API fails to add - & throws an exception, as a result when an exception occurs we should stop execution of code or return.

**Comment by ThejaChoudary** [comment]
> @ThejaChoudary the linked GitHub issue has repro steps can we use this & add as a test case to make sure this scenario is covered to make sure we have no future regressions?

sure will include and send a new PR.

File: src/Websites/Websites/Utilities/WebsitesClient.cs (hunk 1 above)
Comment (ThejaChoudary) [reply src/Websites/Websites/Utilities/WebsitesClient.cs]: I consider the scenario, when they try to add multiple Host Names in a single object and the 1st HN is added successfully where the 2nd HN is failed to add.In this case we should consider either of the following 
1. Remove the successfully added HN and stop the execution. With this the old Host Names will remain as it is.
2. Continue the code execution if at least one custom domain is added and remove the old custom domains.

Here I choose the 2nd option. Please suggest.

**Comment by markcowl** [comment]
@panchagnula Can you review the responses and comment or sign off?

**Comment by panchagnula** [comment]
closing this since this is covered by https://github.com/Azure/azure-powershell/pull/10972

**Pull request closed without merging**